F tree structure

cw-bhanunadar opened the pull request.

Commit:
Pull Request component has been added

cw-bhanunadar committed:
added Favourite Repo Component

edwinclement08 commented on src/utils/Logo.js:
```diff
@@ -0,0 +1,13 @@
+export default function Logo(props) {
+  return (
+    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 841.9 595.3" width="841.9" height="595.3" {...props}>
+      <g fill="#61DAFB">
```
Dude, why don't you just use `import logo.svg`

cw-bhanunadar replied:
removed

edwinclement08 commented on src/utils/PullRequestIcon.js:
```diff
@@ -0,0 +1,14 @@
+  return (
+    <svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 48 48" style={{enableBackground: 'new 0 0 48 48'}} xmlSpace="preserve" width={48} height={48} {...props}>
+      <line style={{fill: 'none', stroke: '#CFD8DC', strokeWidth: 2, strokeMiterlimit: 10}} x1={12} y1={39} x2={12} y2={9} />
+      <path style={{fill: 'none', stroke: '#CFD8DC', strokeWidth: 2, strokeMiterlimit: 10}} d="M21,8h7c4.418,0,8,3.582,8,8v22" />
```
Same here, use `import pr.svg`

cw-bhanunadar replied:
removed

edwinclement08 commented on src/utils/BookmarkIcon.js:
```diff
@@ -0,0 +1,22 @@
+      <path fill="#f15b6c" d="M71 19L29 19 29 85 50 71.864 71 85z" />
+      <path fill="#f37e98" d="M68,22.5v56.177c0,0.393-0.432,0.632-0.765,0.424L50,68.32L32.765,79.101C32.432,79.31,32,79.07,32,78.677V22.5c0-0.276,0.224-0.5,0.5-0.5h35C67.776,22,68,22.224,68,22.5z" />
+      <path fill="#fef6aa" d="M50 28A11 11 0 1 0 50 50A11 11 0 1 0 50 28Z" />
+      <path fill="#f9e65c" d="M55.77,38.35l-2.31,2.35L54,44.01c0.06,0.3-0.06,0.6-0.31,0.79c-0.13,0.15-0.54,0.23-0.84,0.08L50,43.32l-2.83,1.55c-0.11,0.07-0.25,0.1-0.38,0.1c-0.17,0-0.34-0.06-0.48-0.16c-0.25-0.2-0.37-0.5-0.31-0.81l0.54-3.3l-2.31-2.35c-0.2-0.21-0.28-0.53-0.2-0.81c0.12-0.32,0.35-0.52,0.64-0.57l3.19-0.48l1.41-3c0.29-0.57,1.18-0.56,1.45-0.01l1.42,3.01l3.17,0.48c0.31,0.05,0.54,0.25,0.65, ...
```
Same here, import.svg

cw-bhanunadar replied:
removed

edwinclement08 commented on src/css/treeStructure.css:
```diff
@@ -0,0 +1,22 @@
+}
+.sideBar{
+    /* height: 100%;
+  width: 400px;
```
Add newline

edwinclement08 commented on src/containers/SideBar.js:
```diff
@@ -0,0 +1,51 @@
+                )}
+            </>
+        );
+    }
```
newline

edwinclement08 commented on src/containers/SideBar.js:
```diff
@@ -0,0 +1,51 @@
+        favourite.map((value) => {
+            row.push(<FavouriteRepo name={value} />);
+        });
+        return row;
```
Can you re-write this with react hooks. There is just one tiny state(boolean)

edwinclement08 commented on src/containers/SideBar.js:
```diff
@@ -0,0 +1,51 @@
+        }
+    }
+    renderFavourite = () => {
+        console.log("Inside Render Favourites");
```
Remove the Logging statements please.

cw-bhanunadar replied:
done

edwinclement08 commented on src/containers/SideBar.js:
```diff
@@ -0,0 +1,51 @@
+export class SideBar extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
```
Also, lets stick to `2 space` indentation

edwinclement08 commented on src/utils/BookmarkIcon.js:
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+export default function BookmarkIcon(props) {
```
Icons should be kept in a separate folder(../assets/images/icons). You would need to create that.
Also, assets should be out of `src` folder.

cw-bhanunadar replied:
moved to assets

edwinclement08 commented on src/utils/Constants.js:
```diff
@@ -0,0 +1 @@
+export const PULL_REQUEST_ICON_URL = "https://visualpharm.com/assets/817/Pull%20Request-595b40b85ba036ed117da8b8.svg";
```
Are you using this, if so, download the svg and store it into assets, otherwise the load times for extension will depend on network.

cw-bhanunadar replied:
removed

edwinclement08 commented on src/components/FavouriteRepo.js:
```diff
@@ -0,0 +1,13 @@
+            </div>
+        </>
+    );
+}
```
newline

cw-bhanunadar committed:
code review changes

- Moved icons to assets folder
- Made container functional component
- Added Prettier and Eslint

cw-bhanunadar committed:
Removed svg to react package

cw-bhanunadar commented:
I have added extra space to all file but have to check why not showing in PR
